### PR TITLE
Various Travis changes, integration of pkgdown theme

### PR DESCRIPTION
**Pull request by pat-s**
#### Changes

* Integration of `pkgdown` assets into repo (all `pkgdown/*` files). With this we do not anymore rely on my fork.
* Greater use of `tic`: Easier setup of travis stages, more flexibility with env variables
* PDF tutorial is now deployed to branch `tutorial_pdf` and linked from there. This prevents cluttering the master branch with ci commits. 
* Search functionality of `pkgdown` site via [docsearch](https://community.algolia.com/docsearch/)

#### Outlook

I will write a wiki entry explaining the detailed Travis steps and the integration of `tic` so that it becomes more clear for the future. 

We will soon have the possibility to share the cache also across different envs on travis: https://github.com/travis-ci/travis-build/pull/1265#issuecomment-376713016.
This is important for us as currently the warmup cache does not apply to the tutorial builds due to the differing env variable.

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -3,6 +3,18 @@ url: https://mlr-org.github.io/mlr
 template:
   params:
     bootswatch: cosmo
+  path: pkgdown/templates
+  assets: pkgdown/assets
+  default_assets: false
+  docsearch:
```
Comment (jayhesselberth) [inline _pkgdown.yml]: The docsearch info should be *under* params, not at the same level:

```yaml
template:
  params:
    docsearch:
      api_key: e300ecafdf04fe1199e3339c825ce7d0
      index_name: mlr

```

Comment (pat-s) [reply _pkgdown.yml]: Thanks!

**Comment by larskotthoff** [comment]
Thanks, merging.

File: _pkgdown.yml (hunk 1 above)
Comment (jayhesselberth) [reply _pkgdown.yml]: HTH. I think you'll also have to update your custom templates to get the search bar.

**Pull request merged**